service/src/registry.ts

```diff
@@ -523,6 +558,53 @@ export class Registry {
     });
   }
 
+  /**
+   * @method            module:azure-iothub.Registry#updateModuleTwin
+   * @description         Updates the Twin of a specific module with the given patch.
+   *
+   * @param {String}      deviceId    The device identifier.
+   * @param {String}      moduleId    The module identifier
+   * @param {Object}      patch       The desired properties and tags to patch the module twin with.
+   * @param {string}      etag        The latest etag for this module twin or '*' to force an update even if
+   *                                  the module twin has been updated since the etag was obtained.
+   * @param {Function}    done        The callback that will be called with either an Error object or
+   *                                  the module twin instance.
+   * @throws {ReferenceError}         If the deviceId, moduleId, patch, etag, or done argument is falsy.
+   */
+  updateModuleTwin(deviceId: string, moduleId: string, patch: any, etag: string, done: Registry.ResponseCallback): void {
+    /*Codes_SRS_NODE_IOTHUB_REGISTRY_18_004: [The `updateModuleTwin` method shall throw a `ReferenceError` exception if `deviceId`, `moduleId`, `patch`, `etag`,or `done` is falsy. ]*/
+    if (!deviceId) throw new ReferenceError('Argument \'deviceId\' cannot be falsy');
+    if (!moduleId) throw new ReferenceError('Argument \'moduleId\' cannot be falsy');
+    if (!patch) throw new ReferenceError('Argument \'patch\' cannot be falsy');
+    if (!etag) throw new ReferenceError('Argument \'etag\' cannot be falsy');
+    if (!done) throw new ReferenceError('Argument \'done\' cannot be falsy');
```

Comment:
>(!done) [](start = 7, length = 7)

Don't think we should require this.

Reply:
it may become optional when we switch to our APIs to promises. 

---
In reply to: [176867725](https://github.com/Azure/azure-iot-sdk-node/pull/256#discussion_r176867725) [](ancestors = 176867725)